#48 Added registration validation

ONourry opened the pull request:
There should be no dependency issues this time around. Don't forget to run npm install to add the new dependencies.

pbgnz commented:
Your FE validation of the register form works nicely. 

However, you need to keep the bootstrap components so that the form is centered and responsive.

jacobrs commented:
I agree with Pablo, it needs to keep the bootstrap components.

ONourry commented:
I corrected the alignement of the inputs and made them a bit larger. It looks better this way

pbgnz commented on client/modules/User/components/UserRegistrationForm/UserRegistrationForm.css:
```diff
@@ -31,3 +31,8 @@ a:hover {
+
+label, input{
+  vertical-align:middle;
+  width:50%;
```
what happens when you put `min-width:50%`?

ONourry replied:
I re-did it using the bootstrap grid, that should make it more consistent

pbgnz commented on client/modules/User/components/UserRegistrationForm/UserRegistrationForm.js:
```diff
@@ -30,38 +36,85 @@ export class UserRegistrationForm extends Component {
+                  onFocus={this.removeApiError}
+                  className="form-control"
+                  name="password"
+                  type="text"
```
you should put password for the type here, because I don't thing the user should see the password on the form.

`type="password"`

ONourry replied:
Good point, I forgot about that

ONourry commented:
No issues, I see it as improvements rather than roasting. I'll add the confirm password field tonight. I thought about adding one too but I wanted to stick to the issue as much as possible.

MartinSpasov commented:
Is there any requirements for the password field? I just used "aaa" and It created a new user

ONourry commented:
As of right now there is no clear rule. As long as the field is not empty and that it's the same in both password and confirm password it goes through. Do we want a rule for it or just leave it up to the user?

ONourry commented:
Added the password restrictions. The user must now provide a password containing at least 8 characters, out of which 1 uppercase letter, 1 lowercase letter and 1 digit.

MartinSpasov commented:
Perfect! Good job

jacobrs commented:
It looks like this on my screen, the fields are way too short.

<img width="1280" alt="screen shot 2017-02-14 at 12 20 57 am" src="https://cloud.githubusercontent.com/assets/6231440/22916274/7c7a80a8-f24b-11e6-87ed-1181d8ec5b55.png">

Also when I put the same passwords in both fields I get:
<img width="277" alt="screen shot 2017-02-14 at 12 22 58 am" src="https://cloud.githubusercontent.com/assets/6231440/22916334/d0fd665e-f24b-11e6-9a18-016d30b92064.png">

ONourry commented:
I changed the event to Select instead of Focus. I tried it about 10 times and this one seems to be more consistent since you don't have to re-click on the box to hide the error. As far as the components size go, I looked it up and it seems like a known bug between Windows and OS/Unix where the OS render the content differently (goo.gl/XJ7pjV, goo.gl/ORiw9U). As of right right now, windows user see the same inputs as the one on Martin's screenshot but Unix users see it like Jacob's screenshot. The only way to fix it is to add some css to it but since I don't have a unix system I can't tell if the sizes I try are right or not. I added the input{} field in the UserRegistrationForm.css so if someone using unix or mac could just put the correct values so that the fields look as the ones on Martin's screenshot, i'd appreciate it.

jacobrs commented:
@ONourry I'm not sure I understand. Why do the fields render correctly on the login page on both UNIX and windows? The registration fields used to render correctly before your work on the validation as well. Why would adding validation change the fields rendering behaviour? That doesn't make sense to me. I'm sure I'm missing something or not seeing how this works 100%.
